server/src/main/java/org/cloudfoundry/identity/uaa/oauth/UaaTokenServices.java

```diff
@@ -301,7 +299,7 @@ public OAuth2AccessToken refreshAccessToken(String refreshTokenValue, TokenReque
             }
             String newRevocableHashSignature = UaaTokenUtils.getRevocableTokenSignature(client, clientSecretForHash, user);
             if (!revocableHashSignature.equals(newRevocableHashSignature)) {
-                throw new TokenRevokedException(refreshTokenValue);
+                throw new TokenRevokedException("Invalid refresh token: revocable signature mismatch");
```

Comment:
here maybe the id could be logged to that it could be found in revocable DB tablöe